src/main/java/com/artipie/asto/google/GoogleStorage.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.asto.google;
+
+import com.artipie.asto.Content;
+import com.artipie.asto.Key;
+import com.artipie.asto.Remaining;
+import com.artipie.asto.Storage;
+import com.artipie.asto.Transaction;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.StorageOptions;
+import io.reactivex.Flowable;
+import java.io.ByteArrayOutputStream;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Storage that holds data in Google storage.
+ *
+ * @since 0.1
+ */
+@SuppressWarnings({"PMD.UnusedPrivateField", "PMD.SingularField"})
+public final class GoogleStorage implements Storage {
+
+    /**
+     * Google storage client.
+     */
+    private final com.google.cloud.storage.Storage client;
+
+    /**
+     * Bucket name.
+     */
+    private final String bucket;
+
+    /**
+     * Ctor.
+     *
+     * @param bucket Bucket name
+     */
+    public GoogleStorage(final String bucket) {
+        this(StorageOptions.getDefaultInstance(), bucket);
+    }
+
+    /**
+     * Ctor.
+     * @param options Options for Google storage client
+     * @param bucket Bucket name
+     */
+    public GoogleStorage(final StorageOptions options, final String bucket) {
+        this.client = options.getService();
+        this.bucket = bucket;
+    }
+
+    @Override
+    public CompletableFuture<Boolean> exists(final Key key) {
+        throw new UnsupportedOperationException();
```

Comment:
Better to return a future, which completes with an error

Reply:
@Sammers21 it does not matter at this point, because in next release (hopefully) it will be implemented completely.

Reply:
@Guseyn , good